### PR TITLE
Support compilation for __aarch64__.

### PR DESCRIPTION
**Pull request by achichinov**
The test coredumper_unittest fails and not fixed yet.

To support aarch64 copied linux_syscall_support.h from Chromium LSS
https://chromium.googlesource.com/linux-syscall-support

Modified linux_syscall_support.h to fix a redefinition of
struct kernel_dirent64.

LSS for aarch64 does not provide support for sys_sigaction,
sys_sigpending, sys_sigsuspend.
Replaced these calls with sys_rt_ versions and updated their
parameters list.

Updated parameters list of sys_prctl
Un-track libtool (it is generated by configure)
The linuxthreads.c is compiled as a C file, it makes the compiler
skip #ifdef __cplusplus on line 592 and selects the #else branch.
This triggers an error "sys0_waitpid is undefined."
Replaced sys0_waitpid with sys_waitpid

Added includes in thread_lister.c to fix a warning
"implicit declaration of function ‘getpid’"

### CODE REVIEW

File: src/linux_syscall_support_unittest.cc [hunk 1]
```diff
@@ -140,42 +140,43 @@ static void SigAction(int signum, siginfo_t *si, void *arg) {
 
 static void Sigaction() {
   puts("Sigaction...");
+  const size_t kSigsetSize = sizeof(struct kernel_sigset_t);
   int signum       = SIGPWR;
   for (int info = 0; info < 2; info++) {
     signaled         = 0;
     struct kernel_sigaction sa = ZERO_SIGACT, old, orig;
-    CHECK(!sys_sigaction(signum, NULL, &orig));
+    CHECK(!sys_rt_sigaction(signum, NULL, &orig, kSigsetSize));
```
Comment (madscientist) [inline src/linux_syscall_support_unittest.cc]: Just curious if we've verified that this form of syscall is supported in the older kernels we support: we still say we'll run with RHEL 7.5 for example as our oldest kernel.

File: src/linuxthreads.c [hunk 2]
```diff
@@ -597,10 +598,10 @@ int ListAllProcessThreads(void *parameter,
       #define SYS_ERRNO        my_errno
       #define SYS_INLINE       inline
       #define SYS_PREFIX       0
-      #undef  SYS_LINUX_SYSCALL_SUPPORT_H
-      #include "linux_syscall_support.h"
+      // #undef  SYS_LINUX_SYSCALL_SUPPORT_H
+      // #include "linux_syscall_support.h"
```
Comment (madscientist) [inline src/linuxthreads.c]: If this is not needed why don't we just remove it?

Was it ever needed?  If it was before, why is it not needed now?

If you want to keep this here commented out please add some kind of comment about why it's commented out and why someone might want to un-comment it.

File: src/linux_syscall_support_unittest.cc (hunk 1 above)
Comment (achichinov) [reply src/linux_syscall_support_unittest.cc]: The sources of RHEL 7.5 (x86-32 and x86-64) contain the declarations of these sys_rt_ functions.
I hope arm a64 will also contain these declarations.

File: src/linuxthreads.c (hunk 2 above)
Comment (achichinov) [reply src/linuxthreads.c]: I have fixed the error "redefinition of ‘struct kernel_dirent64’". and these lines are not commented out anymore.
I will update the commit message again.

**Pull request closed without merging**